### PR TITLE
Merge Foundry migration to main

### PR DESCRIPTION
**Pull request by wschwab**
This PR migrates the full FX-Portal Foundry migration to main, including a complete test suite and proof generation script.

### CODE REVIEW

**Comment by DhairyaSethi** [comment]
Dissecting this PR for QoL
Main Contracts were affected in the following PRs: 
#54 (Lint and Comments for Slither; no bytecode change), 
#58 #61 (Mintable NFT fixes, the following contracts have changes in bytecode & need to be redeployed: FxMintableERC721RootTunnel, FxMintableERC721ChildTunnel, FxMintableERC1155RootTunnel, FxMintableERC1155ChildTunnel; removal from [static](https://github.com/maticnetwork/static/pull/110))

Unit, integration, & invariant test suite (does not affect main contracts)
#53 (foundry setup; FxRoot, FxChild) 
#55 (ERC20 Invariant + Unit) #57 (Rest Invariant + Unit) 
#59 (Hardhat Integration Tests, Proof Generation Script) 
#60 (Combined Coverage of HH & Foundry)

**Review by dev1644 — APPROVED** [review]

**Review by QEDK — APPROVED** [review]
LGTM.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,42 +4,84 @@ on:
   push:
     branches: [main]
   pull_request:
-    branches: [main]
+    branches: [main, dev/foundry]
```
Comment (QEDK) [inline .github/workflows/ci.yml:7]: @DhairyaSethi can you create another PR that changes this to dev once merged it, so fork main -> dev, then push the branch with branch as dev.

Comment (DhairyaSethi) [reply .github/workflows/ci.yml:7]: #64

**Pull request merged**